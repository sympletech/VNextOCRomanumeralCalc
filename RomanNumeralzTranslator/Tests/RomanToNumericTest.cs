using System;
using System.Linq;
using ApprovalTests;
using ApprovalTests.Reporters;
using ApprovalUtilities.Utilities;
using Microsoft.VisualStudio.TestTools.UnitTesting;

namespace RomanNumeralzTranslator
{
    [UseReporter(typeof(DiffReporter))]

    [TestClass]
    public class RomanToNumericTest
    {
        
        [TestMethod]
        public void RomanToDecimalTest()
        {
            string[] romanDigits = new[]
            {
                "I", "II", "III", "IV", "V", "VI", "VII", "VIII","IX","X",
                "XI", "XII", "XIII", "XIV", "XV", "XVI", "XVII", "XVIII","XIX","XX",
                "XXX","XXXIX","", "XL", "XLIX", "L", "l", "XC", "C",
                "XLVIII"
                
            };
            var results = romanDigits.Select(x => new {key = x, value = RomanNumeralTranslator.Translate(x)});
            ApprovalTests.Approvals.VerifyAll(results, "");
        }

        [TestMethod]
        public void RomanToDecimalExceptionTest()
        {
            var ex = ExceptionUtilities.GetException(() => RomanNumeralTranslator.Translate("Z"));
            Approvals.Verify(ex);
        }

        [TestMethod]
        public void BadDigitsTest()
        {
            string[] badDigits = {"IIX","IIZ","MIM", "MOM", "IVI", "HIV","XXVX", "VV","LL","LLCOOLJ"};
            var ex =
                badDigits.Select(
                    x =>
                        new
                        {
                            Key = x,
                            Value = ExceptionUtilities.GetException(() => RomanNumeralTranslator.Translate(x))
                        });
            Approvals.VerifyAll(ex,"");
        }
    }
}