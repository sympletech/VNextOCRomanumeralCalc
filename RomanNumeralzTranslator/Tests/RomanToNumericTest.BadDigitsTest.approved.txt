[0] = { Key = IIX, Value = System.ArgumentException: Invalid
   at RomanNumeralzTranslator.RomanNumeralTranslator.Translate(String roman) in c:\Users\Daniel\Documents\Visual Studio 2013\Projects\RomanNumeralzTranslator\RomanNumeralzTranslator\Code\RomanNumeralTranslator.cs:line 61
   at RomanNumeralzTranslator.RomanToNumericTest.<>c__DisplayClass7.<BadDigitsTest>b__5() in c:\Users\Daniel\Documents\Visual Studio 2013\Projects\RomanNumeralzTranslator\RomanNumeralzTranslator\Tests\RomanToNumericTest.cs:line 48
   at ApprovalUtilities.Utilities.ExceptionUtilities.GetException(Action action) in c:\code\ApprovalTests.Net\ApprovalUtilities\Utilities\ExceptionUtilities.cs:line 45 }
[1] = { Key = IIZ, Value = System.ArgumentException: 1Z not a valid Roman numeral
   at RomanNumeralzTranslator.RomanNumeralTranslator.ThrowOnInvalidToken(String token) in c:\Users\Daniel\Documents\Visual Studio 2013\Projects\RomanNumeralzTranslator\RomanNumeralzTranslator\Code\RomanNumeralTranslator.cs:line 79
   at RomanNumeralzTranslator.RomanNumeralTranslator.Translate(String roman) in c:\Users\Daniel\Documents\Visual Studio 2013\Projects\RomanNumeralzTranslator\RomanNumeralzTranslator\Code\RomanNumeralTranslator.cs:line 69
   at RomanNumeralzTranslator.RomanToNumericTest.<>c__DisplayClass7.<BadDigitsTest>b__5() in c:\Users\Daniel\Documents\Visual Studio 2013\Projects\RomanNumeralzTranslator\RomanNumeralzTranslator\Tests\RomanToNumericTest.cs:line 48
   at ApprovalUtilities.Utilities.ExceptionUtilities.GetException(Action action) in c:\code\ApprovalTests.Net\ApprovalUtilities\Utilities\ExceptionUtilities.cs:line 45 }
[2] = { Key = MIM, Value = System.ArgumentException: M not a valid Roman numeral
   at RomanNumeralzTranslator.RomanNumeralTranslator.ThrowOnInvalidToken(String token) in c:\Users\Daniel\Documents\Visual Studio 2013\Projects\RomanNumeralzTranslator\RomanNumeralzTranslator\Code\RomanNumeralTranslator.cs:line 79
   at RomanNumeralzTranslator.RomanNumeralTranslator.Translate(String roman) in c:\Users\Daniel\Documents\Visual Studio 2013\Projects\RomanNumeralzTranslator\RomanNumeralzTranslator\Code\RomanNumeralTranslator.cs:line 49
   at RomanNumeralzTranslator.RomanToNumericTest.<>c__DisplayClass7.<BadDigitsTest>b__5() in c:\Users\Daniel\Documents\Visual Studio 2013\Projects\RomanNumeralzTranslator\RomanNumeralzTranslator\Tests\RomanToNumericTest.cs:line 48
   at ApprovalUtilities.Utilities.ExceptionUtilities.GetException(Action action) in c:\code\ApprovalTests.Net\ApprovalUtilities\Utilities\ExceptionUtilities.cs:line 45 }
[3] = { Key = MOM, Value = System.ArgumentException: MOM not a valid Roman numeral
   at RomanNumeralzTranslator.RomanNumeralTranslator.ThrowOnInvalidToken(String token) in c:\Users\Daniel\Documents\Visual Studio 2013\Projects\RomanNumeralzTranslator\RomanNumeralzTranslator\Code\RomanNumeralTranslator.cs:line 79
   at RomanNumeralzTranslator.RomanNumeralTranslator.Translate(String roman) in c:\Users\Daniel\Documents\Visual Studio 2013\Projects\RomanNumeralzTranslator\RomanNumeralzTranslator\Code\RomanNumeralTranslator.cs:line 49
   at RomanNumeralzTranslator.RomanToNumericTest.<>c__DisplayClass7.<BadDigitsTest>b__5() in c:\Users\Daniel\Documents\Visual Studio 2013\Projects\RomanNumeralzTranslator\RomanNumeralzTranslator\Tests\RomanToNumericTest.cs:line 48
   at ApprovalUtilities.Utilities.ExceptionUtilities.GetException(Action action) in c:\code\ApprovalTests.Net\ApprovalUtilities\Utilities\ExceptionUtilities.cs:line 45 }
[5] = { Key = HIV, Value = System.ArgumentException: H not a valid Roman numeral
   at RomanNumeralzTranslator.RomanNumeralTranslator.ThrowOnInvalidToken(String token) in c:\Users\Daniel\Documents\Visual Studio 2013\Projects\RomanNumeralzTranslator\RomanNumeralzTranslator\Code\RomanNumeralTranslator.cs:line 79
   at RomanNumeralzTranslator.RomanNumeralTranslator.Translate(String roman) in c:\Users\Daniel\Documents\Visual Studio 2013\Projects\RomanNumeralzTranslator\RomanNumeralzTranslator\Code\RomanNumeralTranslator.cs:line 49
   at RomanNumeralzTranslator.RomanToNumericTest.<>c__DisplayClass7.<BadDigitsTest>b__5() in c:\Users\Daniel\Documents\Visual Studio 2013\Projects\RomanNumeralzTranslator\RomanNumeralzTranslator\Tests\RomanToNumericTest.cs:line 48
   at ApprovalUtilities.Utilities.ExceptionUtilities.GetException(Action action) in c:\code\ApprovalTests.Net\ApprovalUtilities\Utilities\ExceptionUtilities.cs:line 45 }
[6] = { Key = XXVX, Value = System.ArgumentException: Invalid
   at RomanNumeralzTranslator.RomanNumeralTranslator.Translate(String roman) in c:\Users\Daniel\Documents\Visual Studio 2013\Projects\RomanNumeralzTranslator\RomanNumeralzTranslator\Code\RomanNumeralTranslator.cs:line 61
   at RomanNumeralzTranslator.RomanToNumericTest.<>c__DisplayClass7.<BadDigitsTest>b__5() in c:\Users\Daniel\Documents\Visual Studio 2013\Projects\RomanNumeralzTranslator\RomanNumeralzTranslator\Tests\RomanToNumericTest.cs:line 48
   at ApprovalUtilities.Utilities.ExceptionUtilities.GetException(Action action) in c:\code\ApprovalTests.Net\ApprovalUtilities\Utilities\ExceptionUtilities.cs:line 45 }
